### ISSUE TITLE
Invalid implementation of state verifier for claimable balances 

### ISSUE THREAD

**Issue by abuiles** [stellar/go#2934]
I got the following error running locally 

```
ERRO[2020-08-19T20:44:20.763-05:00] STATE IS INVALID!                             err="addAccountsToStateVerifier failed: Entry does not match the fetched entry. Expected: AAAJ+wAAAAAAAAAAc3b96I5M1hzA+ylKF4az8dBh9fLxyldGX6qTIhG5RtYN4LaykaHrwAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAgAAAAAAAAAAAAAAAA= (pretransform = AAAJ+wAAAAAAAAAAc3b96I5M1hzA+ylKF4az8dBh9fLxyldGX6qTIhG5RtYN4LaykaHrwAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAgAAAAAAAAAAAAAAAA=), actual: AAAJ+wAAAAAAAAAAc3b96I5M1hzA+ylKF4az8dBh9fLxyldGX6qTIhG5RtYN4LaykaHrwAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==" pid=27045 service=expingest
```

**Comment by bartekn** [comment]
Maybe it's because of the sponsors extension?

**Comment by abuiles** [comment]
@bartekn confirmed -- it's because of the account v2 extensions --we need to do something along the following lines

```lang=diff
diff --git a/services/horizon/internal/expingest/verify.go b/services/horizon/internal/expingest/verify.go
index 93cda4ef..2e61961c 100644
--- a/services/horizon/internal/expingest/verify.go
+++ b/services/horizon/internal/expingest/verify.go
@@ -301,6 +301,7 @@ func addAccountsToStateVerifier(verifier *verify.StateVerifier, q history.Ingest
 
 	masterWeightMap := make(map[string]int32)
 	signersMap := make(map[string][]xdr.Signer)
+	sponsoringSignersMap := make(map[string][]xdr.SponsorshipDescriptor)
 	for _, row := range signers {
 		if row.Account == row.Signer {
 			masterWeightMap[row.Account] = row.Weight
@@ -312,6 +313,12 @@ func addAccountsToStateVerifier(verifier *verify.StateVerifier, q history.Ingest
 					Weight: xdr.Uint32(row.Weight),
 				},
 			)
+			if !row.Sponsor.IsZero() {
+				sponsoringSignersMap[row.Account] = append(
+					sponsoringSignersMap[row.Account],
+					xdr.SponsorshipDescriptor(xdr.MustAddressPtr(row.Sponsor.String)),
+				)
+			}
 		}
 	}
 
@@ -356,6 +363,14 @@ func addAccountsToStateVerifier(verifier *verify.StateVerifier, q history.Ingest
 						Buying:  xdr.Int64(row.BuyingLiabilities),
 						Selling: xdr.Int64(row.SellingLiabilities),
 					},
+					Ext: xdr.AccountEntryExtensionV1Ext{
+						V: 2,
+						V2: &xdr.AccountEntryExtensionV2{
+							NumSponsored:        xdr.Uint32(row.NumSponsored),
+							NumSponsoring:       xdr.Uint32(row.NumSponsoring),
+							SignerSponsoringIDs: sponsoringSignersMap[row.AccountID],
+						},
+					},
 				},
 			},
 		}

```

**Comment by bartekn** [comment]
Fixed in #2945.
